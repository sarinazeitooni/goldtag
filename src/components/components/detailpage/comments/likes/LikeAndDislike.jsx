import React, {useState} from 'react';
import '../style/like-and-dislike.scss';
import {useDispatch, useSelector} from "react-redux";
import ThumbDownOutlinedIcon from '@material-ui/icons/ThumbDownOutlined';
import ThumbUpAltOutlinedIcon from '@material-ui/icons/ThumbUpAltOutlined';
import actionTypes from "../../../../redux/actionTypes/actionTypes";
function LikeAndDislike({like,dislike}){
    localStorage.setItem('like' , like);
    localStorage.setItem('dislike' , dislike);
    const dispatch = useDispatch();
    let likes = useSelector((state) => state.likes.like);
    let dislikes = useSelector((state) => state.likes.dislike);
    function Dispatch(action){
        dispatch({
            type : action
        })
    }
    function addDislike(){
        Dispatch(actionTypes.Like);
    }
    function addLike(){
        Dispatch(actionTypes.Dislike);
    }
    return(
        <div className='like-dislike-container'>
            <button className='like-buttons' onClick={addDislike}>
                <ThumbUpAltOutlinedIcon/>
                <span className='counter-container'>{likes}</span>
            </button>
            <button className='like-buttons' onClick={addLike}>
                <ThumbDownOutlinedIcon/>
                <span className='counter-container'>{dislikes}</span>
            </button>
        </div>
    )
}
export default LikeAndDislike;