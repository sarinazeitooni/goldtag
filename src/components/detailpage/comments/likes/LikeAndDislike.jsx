import React from 'react';
import '../style/like-and-dislike.scss';
import {useDispatch, useSelector} from "react-redux";
import ThumbDownOutlinedIcon from '@material-ui/icons/ThumbDownOutlined';
import ThumbUpAltOutlinedIcon from '@material-ui/icons/ThumbUpAltOutlined';
import actionTypes from "../../../../redux/actionTypes/actionTypes";
function LikeAndDislike({id}){
    const dispatch = useDispatch();
    let likes = useSelector((state) => state.likes.like);
    let dislikes = useSelector((state) => state.likes.dislike);
    function Dispatch(action , e){
        dispatch({
            type : action,
            id : e
        })
    }
    function addDislike(e){
        Dispatch(actionTypes.Like , e);
        console.log(e,"dislike");
    }
    function addLike(e){
        Dispatch(actionTypes.Dislike , e);
        console.log(e,"like");
   }
    return(
        <div className='like-dislike-container'>
            <button className='like-buttons' onClick={()=>{addDislike(id)}}>
                <ThumbUpAltOutlinedIcon/>
                <span className='counter-container'>{likes}</span>
            </button>
            <button className='like-buttons' onClick={()=>{addLike(id)}}>
                <ThumbDownOutlinedIcon/>
                <span className='counter-container'>{dislikes}</span>
            </button>
        </div>
    )
}
export default LikeAndDislike;