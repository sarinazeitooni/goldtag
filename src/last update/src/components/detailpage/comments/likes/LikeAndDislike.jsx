import React, {useState} from 'react';
import '../style/like-and-dislike.scss';
import {useDispatch, useSelector} from "react-redux";
import ThumbDownOutlinedIcon from '@material-ui/icons/ThumbDownOutlined';
import ThumbUpAltOutlinedIcon from '@material-ui/icons/ThumbUpAltOutlined';
import actionTypes from "../../../../redux/actionTypes/actionTypes";
import ls from 'local-storage';
function LikeAndDislike({likeId,like,dislike}){
    const dispatch = useDispatch();
    const likeItem = useSelector((state) => state.likes.likes.id);
    let likeNum = useSelector((state) => state.likes.likes.id.like)
    let dislikeNum = useSelector((state) => state.likes.dislikes.id.dislike)
    const dislikeItem = useSelector((state) => state.likes.dislikes.id);
    // let likes = (likeItem == likeId) ? likeNum : 0;
    // let dislikes =(dislikeItem == likeId) ? dislikeNum : 0;
    function Dispatch(action){
        dispatch({
            type : action ,
            payload: {
                likeId
            }
        })
    }
    function addDislike(){
        Dispatch(actionTypes.Like);
    }
    function addLike(){
        Dispatch(actionTypes.Dislike);
    }
    return(
        <div className='like-dislike-container'>
            <button className='like-buttons' onClick={addDislike}>
                <ThumbUpAltOutlinedIcon/>
                <span className='counter-container'>{likeNum}</span>
            </button>
            <button className='like-buttons' onClick={addLike}>
                <ThumbDownOutlinedIcon/>
                <span className='counter-container'>{dislikeNum}</span>
            </button>
        </div>
    )
}
export default LikeAndDislike;