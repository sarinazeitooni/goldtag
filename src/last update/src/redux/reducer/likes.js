const initialState = {
    likes : {
        like : 0,
        dislike : 0
    }
}
export default function likes(state=initialState, action) {
    switch (action.type) {
        case 'Like':
            return {
                ...state,
                likes : {
                    id : action.payload.likeId,
                    like : state.likes.like+1
                }
            };
        case 'Dislike':
            return{
                ...state,
                dislikes : {
                    id : action.payload.likeId,
                    dislike : state.dislikes.dislike+1
                }
            }
        default:
            return state;
    }
}