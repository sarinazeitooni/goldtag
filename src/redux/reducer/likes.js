let initialState = {
    like: 0,
    dislike: 0
}
let id;
let target;
let data;
export default function likes(state = initialState, action) {
    console.log(action , "ACTION");
    switch (action.type) {

        case 'Like':
            id = action.id;
            data = action.data

            return {
                state,
                like: state.like + 1,
                id : action.id
            }
        case 'Dislike':
            id = action.id;
            return {
                ...state,
                dislike: state.dislike + 1,
                id : action.id
            }
        default:
            return state;
    }
}