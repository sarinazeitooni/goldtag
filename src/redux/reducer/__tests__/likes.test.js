import React from 'react';
import {shallow} from 'enzyme';
import likes from "../likes";
describe("like reducer component", () => {
    const container = shallow(<likes/>)
    it("it should render", () => {
        expect(container).toMatchSnapshot();
    });
});