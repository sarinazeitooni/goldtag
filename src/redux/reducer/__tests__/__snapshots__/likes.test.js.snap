// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`like reducer component it should render 1`] = `ShallowWrapper {}`;
